### DISCUSSION TITLE
Add a "Safe Mode" for Joomla

### DISCUSSION THREAD

**Discussion by Hackwar** [joomla/joomla-cms#47017] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
Users are regularly encountering issues with plugins breaking sites after enabling them or during updates and don't know how to disable them again. 

### Describe the solution you'd like
Add a new switch in the configuration.php to enable a "Safe Mode", which disables loading any plugin, resetting the installation to a minimal Joomla installation. The only thing working like before would be the authentication plugins to get back into the backend. This would allow to disable plugins and do additional work, since as far as I can see all changes in the backend should be possible without any plugins. At the same time, the mode would be activated during Joomla updates to prevent update issues.

### Additional context
Thanks to @HLeithner for bringing this idea up.

**Comment by brianteeman** [comment]
This has been suggested before. Can't find the link right now but there were some very loud voices against it. I think it was in a discussion about isSite

**Comment by QuyTon** [comment]
https://github.com/joomla/joomla-cms/issues/18990

**Comment by brianteeman** [comment]
Thanks for the link @QuyTon

**Comment by StefanSTS** [comment]
If safe mode is not an option for Joomla core, maybe someone could write a recovery script in a gist, that you could upload to /administrator.
(Since that might be a risk to the site if it is in plain view, there should be an explanation how to make the access safe, of course.)

It could provide:
- listing all plugins and modules with checkboxes to mark to switch on/off.
- mass checkboxes for types. 
- extension ID ranges.

Then you don't have use the DB to disable all those external extensions.
It might actually safe some time if you can click yourself through those plugins, reload, click, reload...

**Comment by HLeithner** [comment]
Safe Mode is possible it only depends of the expectations what the purpose are.
It would already be enough if we can complete the update process (last step is execute database updates).
Also I'm pretty sure that more then 95% of the joomla websites uses the default joomla login plugins and and even more as fall back for the admin.

**Comment by alikon** [comment]
my take #44765

**Comment by HLeithner** [comment]
Your approach goes into a completely different direction compared what I have in mind an is to invasive. Parts of it would be interesting for plugin manger to disable all plugins except core plugins for failure search.
